platform/api/azure/instance.go

```diff
@@ -147,7 +147,10 @@ func (a *API) CreateInstance(name, userdata, sshkey, resourceGroup, storageAccou
 		return true, nil
 	})
 	if err != nil {
-		a.TerminateInstance(name, resourceGroup)
```

Comment:
You modified TerminateInstance to do what the code here does... So I think you should still call TerminateInstance here?

Reply:
I thought so but the machine object is not created yet and the other option is to have three arguments in the TerminateInstance function.